File: github-actions-backend/test-thebrain-api.js
#!/usr/bin/env node

// Test script for TheBrain API connection
const axios = require('axios');
require('dotenv').config();

async function testTheBrainAPI() {
  console.log('🧠 Testing TheBrain API Connection...\n');
  
  const apiKey = process.env.THEBRAIN_API_KEY;
  const brainId = process.env.THEBRAIN_BRAIN_ID;
  
  if (!apiKey) {
    console.error('❌ THEBRAIN_API_KEY not found in environment variables');
    return false;
  }
  
  if (!brainId) {
    console.error('❌ THEBRAIN_BRAIN_ID not found in environment variables');
    return false;
  }
  
  console.log('📋 Configuration:');
  console.log(`   API Key: ${apiKey.substring(0, 10)}...${apiKey.substring(apiKey.length - 5)}`);
  console.log(`   Brain ID: ${brainId}\n`);
  
  const api = axios.create({
    baseURL: 'https://api.bra.in',
    headers: {
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${apiKey}`
    },
    timeout: 10000
  });
  
  try {
    // Test 1: Get Brain Info
    console.log('1️⃣ Testing Brain Access...');
    const brainResponse = await api.get(`/brains/${brainId}`);
    console.log(`   ✅ Brain Name: ${brainResponse.data.name}`);
    console.log(`   ✅ Home Thought ID: ${brainResponse.data.homeThoughtId}\n`);
    
    // Test 2: Search for existing thoughts
    console.log('2️⃣ Testing Search...');
    const searchResponse = await api.get(`/search/${brainId}`, {
      params: {
        queryText: 'AI',
        maxResults: 5
      }
    });
    console.log(`   ✅ Found ${searchResponse.data.length} results\n`);
    
    // Test 3: Create a test thought
    console.log('3️⃣ Testing Thought Creation...');
    const testThought = {
      name: `API Test - ${new Date().toISOString()}`,
      label: '🔧',
      kind: 1, // Normal
      acType: 0 // Public
    };
    
    const createResponse = await api.post(`/thoughts/${brainId}`, testThought);
    const thoughtId = createResponse.data.id;
    console.log(`   ✅ Created thought with ID: ${thoughtId}\n`);
    
    // Test 4: Add a note
    console.log('4️⃣ Testing Note Creation...');
    const noteContent = `# API Test Note

This is a test note created at ${new Date().toLocaleString()}

## Test Results
- API Connection: ✅ Working
- Brain Access: ✅ Verified
- Thought Creation: ✅ Successful
- Note Creation: ✅ In Progress`;
    
    await api.post(`/notes/${brainId}/${thoughtId}/update`, {
      markdown: noteContent
    });
    console.log('   ✅ Note added successfully\n');
    
    // Test 5: Create a link to home thought
    console.log('5️⃣ Testing Link Creation...');
    if (brainResponse.data.homeThoughtId) {
      await api.post(`/links/${brainId}`, {
        thoughtIdA: brainResponse.data.homeThoughtId,
        thoughtIdB: thoughtId,
        relation: 1, // Child
        name: 'API Test'
      });
      console.log('   ✅ Link created successfully\n');
    }
    
    // Test 6: Clean up - delete the test thought
    console.log('6️⃣ Cleaning up...');
    await api.delete(`/thoughts/${brainId}/${thoughtId}`);
    console.log('   ✅ Test thought deleted\n');
    
    console.log('🎉 All API tests passed successfully!');
    console.log('   TheBrain API integration is working correctly.\n');
    
    return true;
    
  } catch (error) {
    console.error('\n❌ API Test Failed:');
    
    if (error.response) {
      console.error(`   Status: ${error.response.status}`);
      console.error(`   Message: ${error.response.data.error || error.response.statusText}`);
      
      if (error.response.status === 401) {
        console.error('\n   🔑 Authentication failed. Please check:');
        console.error('      - Is your API key correct?');
        console.error('      - Is the API key active?');
        console.error('      - Does it have the right permissions?');
      } else if (error.response.status === 404) {
        console.error('\n   🔍 Resource not found. Please check:');
        console.error('      - Is the Brain ID correct?');
        console.error('      - Do you have access to this brain?');
      }
    } else if (error.request) {
      console.error('   No response received from API');
      console.error('   Check your internet connection');
    } else {
      console.error(`   Error: ${error.message}`);
    }
    
    return false;
  }
}

// Run the test
if (require.main === module) {
  testTheBrainAPI().then(success => {
    process.exit(success ? 0 : 1);
  });
}

module.exports = testTheBrainAPI;

File: github-actions-backend/test-thebrain-api.py
#!/usr/bin/env python3
"""
Test TheBrain API endpoints to find the correct format
"""

import requests
import json
import os

# Test different API configurations
API_KEY = os.getenv('THEBRAIN_API_KEY', '4c8eabea756ab437069c89c46ce7cf00c477591ab9787a526617370d22a4266c')
BRAIN_ID = os.getenv('THEBRAIN_BRAIN_ID', '134f1325-4a8d-46d7-a078-5386c8ab3542')

# Test configurations
test_configs = [
    {
        'name': 'v1 with Bearer',
        'url': f'https://api.thebrain.com/v1/brains/{BRAIN_ID}',
        'headers': {
            'Authorization': f'Bearer {API_KEY}',
            'Content-Type': 'application/json'
        }
    },
    {
        'name': 'No version with Bearer',
        'url': f'https://api.thebrain.com/brains/{BRAIN_ID}',
        'headers': {
            'Authorization': f'Bearer {API_KEY}',
            'Content-Type': 'application/json'
        }
    },
    {
        'name': 'v1 with apikey header',
        'url': f'https://api.thebrain.com/v1/brains/{BRAIN_ID}',
        'headers': {
            'apikey': API_KEY,
            'Content-Type': 'application/json'
        }
    },
    {
        'name': 'v1 with X-API-Key',
        'url': f'https://api.thebrain.com/v1/brains/{BRAIN_ID}',
        'headers': {
            'X-API-Key': API_KEY,
            'Content-Type': 'application/json'
        }
    },
    {
        'name': 'v2 with Bearer',
        'url': f'https://api.thebrain.com/v2/brains/{BRAIN_ID}',
        'headers': {
            'Authorization': f'Bearer {API_KEY}',
            'Content-Type': 'application/json'
        }
    },
    {
        'name': 'API key in URL parameter',
        'url': f'https://api.thebrain.com/v1/brains/{BRAIN_ID}?apikey={API_KEY}',
        'headers': {
            'Content-Type': 'application/json'
        }
    }
]

print("Testing TheBrain API endpoints...")
print("=" * 60)
print(f"Brain ID: {BRAIN_ID}")
print(f"API Key: {API_KEY[:10]}...")
print("=" * 60)

for config in test_configs:
    print(f"\nTesting: {config['name']}")
    print(f"URL: {config['url']}")
    print(f"Headers: {json.dumps({k: v[:20] + '...' if k.lower().endswith('key') or k == 'Authorization' else v for k, v in config['headers'].items()}, indent=2)}")
    
    try:
        response = requests.get(config['url'], headers=config['headers'], timeout=10)
        print(f"Status: {response.status_code}")
        
        if response.status_code == 200:
            print("✅ SUCCESS!")
            print(f"Response: {json.dumps(response.json(), indent=2)}")
            break
        else:
            print(f"❌ Failed")
            if response.text:
                print(f"Response: {response.text[:200]}")
    except Exception as e:
        print(f"❌ Error: {e}")
    
    print("-" * 40)

print("\nTesting complete.")
